### PR TITLE
Hb2

### PR DESCRIPTION
**Pull request by jah701**

### CODE REVIEW

File: src/main/java/com/dev/cinema/model/MovieSession.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+@Table(name = "movie_sessions")
+public class MovieSession {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ManyToOne
+    private Movie movie;
+
+    @ManyToOne
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/model/MovieSession.java:21]: this is for your info, but keep in mind that toOne relations have fetchType eager by default, which is not always quite good

File: src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<MovieSession> getAvailableMovieSessions(Long id, LocalDate date) {
+        LocalDate from = LocalDate.from(date.atStartOfDay());
+        LocalDate to = LocalDate.from(date.atTime(LocalTime.MAX));
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java]: these variables are redundant, just pass parameters setParameter() right away

File: src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> query =
+                    session.createQuery("FROM MovieSession "
+                                    + "WHERE id = :id "
+                                    + "AND showtime "
+                                    + "BETWEEN :from AND :to", MovieSession.class);
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java]:
```suggestion
                    session.createQuery("FROM MovieSession "
                                    + "WHERE id = :id  + AND showtime "
                                    + "BETWEEN :from AND :to", MovieSession.class);
```
don't make code longer unnecessary

File: src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(movieSession);
+            transaction.commit();
+            return movieSession;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add movie session", e);
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java]: let's make message more informative and add some info about entity we were trying to save same with other methods, where applicable

File: src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(cinemaHall);
+            transaction.commit();
+            return cinemaHall;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add movie session", e);
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java]: is message correct?

Comment (jah701) [reply src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java]: Oh it was probabley mixed with previous prs -_- sorry ill fix it now

File: src/main/java/com/dev/cinema/model/MovieSession.java (hunk 1 above)
Comment (jah701) [reply src/main/java/com/dev/cinema/model/MovieSession.java:21]: We've discussed about today :) I've got it now! But it supposed to be acceptable as long as its not toMany right?

Comment (RomanDubovskyi) [reply src/main/java/com/dev/cinema/model/MovieSession.java:21]: it's not critical, just not the best way when we're talking about performance, but it's acceptable

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**